src/Nethermind/Nethermind.Core/Nethermind.Core.csproj

```diff
@@ -13,6 +13,7 @@
     <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" />
     <PackageReference Include="Nethermind.Crypto.SecP256k1" />
     <PackageReference Include="Nethermind.Numerics.Int256" />
+    <PackageReference Include="NonBlocking" />
```

Comment:
In most of our test regular ConcurrentDictionary was better than NonBlocking one. Only in very high contention scenarios non blocking can be better